src/adapter/src/subscribe.rs

```diff
@@ -316,12 +369,38 @@ impl ActiveSubscribe {
 
 /// The reason for removing an [`ActiveSubscribe`].
 #[derive(Debug, Clone)]
-pub enum SubscribeRemovalReason {
-    /// The subscribe completed successfully.
+pub enum ComputeSinkRemovalReason {
+    /// The compute sink completed successfully.
     Finished,
-    /// The subscribe was canceled due to a user request.
+    /// The compute sink was canceled due to a user request.
     Canceled,
-    /// The subscribe was forcibly terminated because an object it depended on
+    /// The compute sink was forcibly terminated because an object it depended on
     /// was dropped.
     DependencyDropped(String),
 }
+
+/// A description of an active copy to from coord's perspective.
+#[derive(Debug)]
+pub struct ActiveCopyTo {
+    /// The user of the session that created the subscribe.
+    pub user: User,
+    /// The connection id of the session that created the subscribe.
+    pub conn_id: ConnectionId,
+    /// The context about the COPY TO statement getting executed.
+    /// Used to eventually call `ctx.retire` on.
+    pub ctx: Option<ExecuteContext>,
+    /// The cluster that the copy to is running on.
+    pub cluster_id: ClusterId,
+    /// All `GlobalId`s that the copy to's expression depends on.
+    pub depends_on: BTreeSet<GlobalId>,
+}
+
+impl ActiveCopyTo {
+    pub(crate) fn process_response(&mut self, response: Result<ExecuteResponse, AdapterError>) {
```

Comment:
This function either should not exist (my preference), or take the owned `self`, making the struct unusable after calling.

Reply:
(Resolving since we addressed this in slack)